polkadot/xcm/xcm-builder/src/routing.rs

```diff
@@ -149,12 +153,20 @@ impl EnsureDelivery for Tuple {
 /// Inspects messages in queues.
 /// Meant to be used in runtime APIs, not in runtimes.
 pub trait InspectMessageQueues {
+	/// Clear the queues, so `Self::get_messages()` can return only the desired events.
```

Comment:
```suggestion
	/// Clear the queues at the beginning of Runtime API call, so that subsequent
	/// `Self::get_messages()` will return only messages generated by said Runtime API call.
```